### PR TITLE
[ACL-242] Add CreditableAt in GetPaymentResult

### PR DESCRIPTION
**Pull request by tl-Roberto-Mancinelli**
Added CreditableAt in `Authorized`, `Executed` and `Settled` case of `GetPayment`.
Added test for GetPayment when status is `Settled`

### CODE REVIEW

File: src/TrueLayer/Payments/Model/GetPaymentResponse.cs [hunk 1]
```diff
@@ -76,19 +76,21 @@ public record Authorizing : PaymentDetails;
 
         /// <summary>
         /// Represents a payment that has been authorized by the end user
+        /// <param name="CreditableAt">The date and time that TrueLayer determined that the payment was ready to be credited</param>
         /// </summary>
         /// <returns></returns>
         [JsonDiscriminator("authorized")]
-        public record Authorized : PaymentDetails;
+        public record Authorized(DateTime? CreditableAt) : PaymentDetails;
```
Comment (dili91) [inline src/TrueLayer/Payments/Model/GetPaymentResponse.cs:83]: Is this correct? It does not seems so based on our public docs? I can't see the field on the `authorized` payment details on the API ref

File: test/TrueLayer.AcceptanceTests/MockBank/MockBankClient.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class MockBankClient
+{
+    private readonly HttpClient _httpClient;
+
+    public MockBankClient(HttpClient httpClient)
+    {
+        _httpClient = httpClient;
+    }
+
+    public async Task<Uri> AuthorisePaymentAsync(
```
Comment (dili91) [inline test/TrueLayer.AcceptanceTests/MockBank/MockBankClient.cs:21]: I think we do something hacky already for similar purposes, for [payments](https://github.com/TrueLayer/truelayer-dotnet/blob/99d7cc1750ef873380b1b2f12409804be5ebd2c1/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L669) and [mandates](https://github.com/TrueLayer/truelayer-dotnet/blob/99d7cc1750ef873380b1b2f12409804be5ebd2c1/test/TrueLayer.AcceptanceTests/MandatesTests.cs#L337)

Could we make this function usable for every resource type (mandates and payments) and reuse this everywhere?
Not needed now, we can do this in another PR

File: src/TrueLayer/Payments/Model/GetPaymentResponse.cs (hunk 1 above)
Comment (tl-Roberto-Mancinelli) [reply src/TrueLayer/Payments/Model/GetPaymentResponse.cs:83]: yeah, it's quite new, probably the docs are not updated, I'm checking with Risk

File: test/TrueLayer.AcceptanceTests/MockBank/MockBankClient.cs (hunk 2 above)
Comment (tl-Roberto-Mancinelli) [reply test/TrueLayer.AcceptanceTests/MockBank/MockBankClient.cs:21]: yeah, I was planning to do a refactor later, we can optimise several part of the tests.
The problem with this [method](https://github.com/TrueLayer/truelayer-dotnet/blob/99d7cc1750ef873380b1b2f12409804be5ebd2c1/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L669) was that returns a GetPaymentResponse.PaymentDetails and we need the specific type of the GetPaymentResponse for this test.

File: src/TrueLayer/Payments/Model/GetPaymentResponse.cs (hunk 1 above)
Comment (tl-Roberto-Mancinelli) [reply src/TrueLayer/Payments/Model/GetPaymentResponse.cs:83]: It's confirmed, we need to update docs

**Review by dili91 — APPROVED** [review]

**Pull request merged**